Assets/Plugins/Android/Core/AndroidFileIOHelper.cs

```diff
@@ -23,35 +23,50 @@
 namespace PlayEveryWare.EpicOnlineServices
 {
     using System;
+    using System.Threading.Tasks;
     using UnityEngine;
+    using UnityEngine.Networking;
 
-    public class AndroidFileIOHelper : MonoBehaviour
+    public class AndroidFileIOHelper
     {
-        public static string ReadAllText(string filePath)
+        public static async Task<string> ReadAllText(string filePath)
         {
-            using (var request = UnityEngine.Networking.UnityWebRequest.Get(filePath))
+            using UnityWebRequest request = UnityWebRequest.Get(filePath);
+            request.timeout = 2; //seconds till timeout
+            UnityWebRequestAsyncOperation operation = request.SendWebRequest();
+
+            while (!operation.isDone)
             {
-                request.timeout = 2; //seconds till timeout
-                request.SendWebRequest();
+                await Task.Yield();
+            }
 
-                //Wait till webRequest completed
-                while (!request.isDone) { }
```

Comment:
haha wow this is a terrifying bit of code. Glad you adjusted it.

Reply:
Yeah, I did that. It sure looks scary. The request has the timeout set to 2 though.
There are definitely better ways to achieve the same thing.